specs/mgt-search.md

```diff
@@ -0,0 +1,182 @@
+
+The following examples shows how to use customized template `suggested-query`.
+
+```html
+<mgt-search>
+    <template data-type="suggested-query">
+        <div>{{query}}</div>
+    </template>
+</mgt-search>
+```
+
+
+## Customized
+Customized entity types must be supported by microsoft graph suggestion API.
+```html
+<mgt-search entity-types="query,file,people,sample1">
+    <template data-type="customized-sample1-header">
+        <div>header hahaha</div>
+    </template>
+
+    <template data-type="customized-sample1">
+        <div>{{this}}</div>
+    </template>
+</mgt-search>
+```
+
+## Proposed Solution
+
+## Example
+
+### Example 1: basic usage
+```<mgt-search></mgt-search>```
+
+### Example 2: get suggestion data and define click and enter key action by adding a listener
+```
+<mgt-search> </mgt-search>
+
+<script>
+
+document.querySelector('mgt-search').addEventListener('onEntityClick', e => {
+    console.log(e.detail);
+});
+
+document.querySelector('mgt-search').addEventListener('onEnterPress', e => {
+  console.log(e.detail);
+});
+
+</script>
+
+```
+
+### Example4: Select entity types & Sort & Suggestion Amount
+It allows customized entity order and max suggestion account by using entity-types properties:
+For example below shows
+Order: query, file, people
+Amount: 2,1,3 ( 3 is default value)
+```html
+  <mgt-search entity-types="query-2, file-1, people"></mgt-search>
+```
+
+```html
+  <mgt-search entity-types="query-2, people"></mgt-search>
+```
+
+```html
+  <mgt-search entity-types="query-2, people, sample1-4"></mgt-search>
+```
+
+## Events
+
+The following events are fired from the component.
+
+| Event | Description |
+| --- | --- |
+| `onEntityClick` | When click an entity , the listener will be trigged. the suggested value of the clicked item as a parameter|
+| `onEnterPress` | When press enter key, the listener will be trigged, it has originalValue( input box value) and suggestedValue (suggestion) as parameters|
+
+## Attributes and Properties
+
+| Attribute | Property | Description |
+| --------- | -------- | ----------- |
+| `entity-types` | `entityTypes` | Suggestion entity types, free combination of query/people/file or other entity types supported by suggestion API, it determines the order,  use ',' to do the segmentation, use '-' to do the amount limitation. for example 'file-2, query-1, people', it menas order: file,query, people.  Entity Types: file, query, people. Amount Limitation: file is 2, query is 1, people is 3 due to the default value is 3 |
+| `other properties` | `other properties` | awaiting for the graph suggestion API onboard. |
+
+## Themes
+### light(default)
+<img src="./images/mgt-search-light.png" width=400/>
+
+### dark
+<img src="./images/mgt-search-dark.png" width=400/>
+
+## CSS custom properties
+
+The `mgt-searach-suggestion` component defines the following CSS custom properties.
+
+```css
+mgt-search {
```

Comment:
There will likely be many more, but we don't have to have them figure out now - let's lock down on the spec and we can move to design stage after